src/gtfs/read.rs

```diff
@@ -859,56 +860,68 @@ where
-                        .get(&transfer.from_stop_id)
-                        .ok_or_else(|| format_err!(
-                            "Problem reading {:?}: from_stop_id={:?} not found",
-                            file,
-                            transfer.from_stop_id
-                        )),
+                let expand_stop_area = |stop_id: &str| -> Result<Vec<&StopPoint>> {
+                    if stop_areas.get(stop_id).is_some() {
+                        let list_stop_points = stop_points
+                            .values()
+                            .filter(|stop_point| stop_point.stop_area_id == stop_id)
+                            .collect();
+                        Ok(list_stop_points)
+                    } else {
+                        stop_points
+                            .get(stop_id)
+                            .ok_or_else(|| {
+                                format_err!(
+                                    "Problem reading {:?}: stop_id={:?} not found",
+                                    file,
+                                    stop_id
+                                )
+                            })
+                            .map(|stop_point| vec![stop_point])
+                    }
+                };
+                let from_stop_points = skip_error_and_log!(
+                    expand_stop_area(transfer.from_stop_id.as_str()),
                     LogLevel::Warn
                 );
-
-                let to_stop_point = skip_error_and_log!(
-                    stop_points
-                        .get(&transfer.to_stop_id)
-                        .ok_or_else(|| format_err!(
-                            "Problem reading {:?}: to_stop_id={:?} not found",
-                            file,
-                            transfer.to_stop_id
-                        )),
+                let to_stop_points = skip_error_and_log!(
+                    expand_stop_area(transfer.to_stop_id.as_str()),
                     LogLevel::Warn
                 );
-
-                let (min_transfer_time, real_min_transfer_time) = match transfer.transfer_type {
-                    TransferType::Recommended => {
-                        let distance = from_stop_point.coord.distance_to(&to_stop_point.coord);
-                        let transfer_time = (distance / 0.785) as u32;
-
-                        (Some(transfer_time), Some(transfer_time + 2 * 60))
+                for from_stop_point in &from_stop_points {
+                    for to_stop_point in &to_stop_points {
+                        let (min_transfer_time, real_min_transfer_time) = match transfer
+                            .transfer_type
+                        {
+                            TransferType::Recommended => {
+                                let distance =
+                                    from_stop_point.coord.distance_to(&to_stop_point.coord);
+                                let transfer_time = (distance / 0.785) as u32;
```

Comment:
to avoid expensive use of cos, sin
```suggestion
                for from_stop_point in &from_stop_points {
                    let approx = from_stop_point.coord.approx();
                    for to_stop_point in &to_stop_points {
                        let (min_transfer_time, real_min_transfer_time) = match transfer
                            .transfer_type
                        {
                            TransferType::Recommended => {
                                let sq_distance = approx.sq_distance_to(&to_stop_point.coord);
                                let transfer_time = (sq_distance.sqrt()  / 0.785) as u32;
```
not tested

Reply:
Nice :-D Should indeed avoid a bunch of probably intensive mathematical operations. Done in fcc177c1c3c17a2c2b1043de8505c682752859f3.